```
# uparmar @ uparmar-mbp in ~/Development/mattermost-govet on git:master
$ cat ../len.go
package main
import "fmt"

func main() {
  s := "asdasd"
  //if len(s) == 0 && len(s) == 1 || len(s) == 3 {
  //  fmt.Println("haha")
  //}
  fmt.Println(len(s) == 0)
}

# uparmar @ uparmar-mbp in ~/Development/mattermost-govet on git:master x [22:52:22]
$ go vet -vettool=mattermost-govet -emptyStrCmp ../len.go
# command-line-arguments
../len.go:10:15: calling len for string and comparing to 0, please compare it to empty string("") instead

# uparmar @ uparmar-mbp in ~/Development/mattermost-govet on git:master x [22:52:35]
$ vim ../len.go

# uparmar @ uparmar-mbp in ~/Development/mattermost-govet on git:master x [22:52:49]
$ cat ../len.go
package main

import "fmt"

func main() {
  s := "asdasd"
  //if len(s) == 0 && len(s) == 1 || len(s) == 3 {
  //  fmt.Println("haha")
  //}
  fmt.Println(len(s) == 1)
}

# uparmar @ uparmar-mbp in ~/Development/mattermost-govet on git:master x [22:53:14]
$ go vet -vettool=mattermost-govet -emptyStrCmp ../len.go

# uparmar @ uparmar-mbp in ~/Development/mattermost-govet on git:master x [22:53:19]
$
```

--- a/emptyStrCmp/emptyStrCmp.go
+++ b/emptyStrCmp/emptyStrCmp.go
@@ -1,11 +1,12 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-package emptystrcmp
+package emptyStrCmp
 
 import (
 	"go/ast"
 	"go/token"
+
 	"golang.org/x/tools/go/analysis"
 )
 
@@ -18,20 +19,34 @@ var Analyzer = &analysis.Analyzer{
 func run(pass *analysis.Pass) (interface{}, error) {
 	for _, file := range pass.Files {
 		ast.Inspect(file, func(node ast.Node) bool {
-			switch node.(type) {
+			switch n := node.(type) {
 			case *ast.BinaryExpr:
-				b := node.(*ast.BinaryExpr)
-				if callExpr, ok := b.X.(*ast.CallExpr); ok {
-					if idt, ok := callExpr.Fun.(*ast.Ident); ok && idt.Name == "len" &&
-						(b.Op == token.EQL || b.Op == token.NEQ) {
-						arg0 := callExpr.Args[0]
-						typ, ok := pass.TypesInfo.Types[arg0]
-						if ok && typ.Type.String() == "string" {
-							if bLit, ok := b.Y.(*ast.BasicLit); ok && bLit.Value == "0" {
-								pass.Reportf(callExpr.Pos(), "calling len for string and comparing to 0, please compare it to empty string(\"\") instead")
-								return false
-							}
+				callExpr, ok := n.X.(*ast.CallExpr)
+				if !ok {
+					return true
+				}
+				idt, ok := callExpr.Fun.(*ast.Ident)
+				if !ok {
+					return true
+				}
+				if idt.Name == "len" && (n.Op == token.EQL || n.Op == token.NEQ) {
+					arg0 := callExpr.Args[0]
+					typ, ok := pass.TypesInfo.Types[arg0]
+					if ok && typ.Type.String() == "string" {
+						bLit, ok := n.Y.(*ast.BasicLit)
+						if !ok {
+							return true
 						}
+						if bLit.Value == "0" {
+							switch n.Op {
+							case token.EQL:
+								pass.Reportf(callExpr.Pos(), "calling len(s) == 0 where s is string, please use s == \"\" instead")
+							case token.NEQ:
+								pass.Reportf(callExpr.Pos(), "calling len(s) != 0 where s is string, please use s != \"\" instead")
+							}
+							return false
+						}
+
 					}
 				}
 			}

--- a/emptyStrCmp/emptyStrCmp.go
+++ b/emptyStrCmp/emptyStrCmp.go
@@ -1,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package emptystrcmp
+
+import (
+	"go/ast"
+	"go/token"
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "emptyStrCmp",
+	Doc:  "check for len(s) == 0 and len(s) != 0 where s is string",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch node.(type) {
+			case *ast.BinaryExpr:
+				b := node.(*ast.BinaryExpr)
+				if callExpr, ok := b.X.(*ast.CallExpr); ok {
+					if idt, ok := callExpr.Fun.(*ast.Ident); ok && idt.Name == "len" &&
+						(b.Op == token.EQL || b.Op == token.NEQ) {
+						arg0 := callExpr.Args[0]
+						typ, ok := pass.TypesInfo.Types[arg0]
+						if ok && typ.Type.String() == "string" {
+							if bLit, ok := b.Y.(*ast.BasicLit); ok && bLit.Value == "0" {
+								pass.Reportf(callExpr.Pos(), "calling len for string and comparing to 0, please compare it to empty string(\"\") instead")
+								return false
+							}
+						}
+					}
+				}
+			}
+			return true
+		})
+	}
+	return nil, nil
+}

--- a/emptyStrCmp/emptyStrCmp.go
+++ b/emptyStrCmp/emptyStrCmp.go
@@ -29,7 +29,7 @@ func run(pass *analysis.Pass) (interface{}, error) {
 				if !ok {
 					return true
 				}
-				if idt.Name == "len" && (n.Op == token.EQL || n.Op == token.NEQ) {
+				if idt.Name == "len" && (n.Op == token.EQL || n.Op == token.NEQ || n.Op == token.GTR) {
 					arg0 := callExpr.Args[0]
 					typ, ok := pass.TypesInfo.Types[arg0]
 					if ok && typ.Type.String() == "string" {
@@ -43,10 +43,11 @@ func run(pass *analysis.Pass) (interface{}, error) {
 								pass.Reportf(callExpr.Pos(), "calling len(s) == 0 where s is string, please use s == \"\" instead")
 							case token.NEQ:
 								pass.Reportf(callExpr.Pos(), "calling len(s) != 0 where s is string, please use s != \"\" instead")
+							case token.GTR:
+								pass.Reportf(callExpr.Pos(), "calling len(s) > 0 where s is string, please use s != \"\" instead")
 							}
 							return false
 						}
-
 					}
 				}
 			}

--- a/main.go
+++ b/main.go
@@ -2,7 +2,7 @@ package main
 
 import (
 	"github.com/mattermost/mattermost-govet/apiAuditLogs"
-	"github.com/mattermost/mattermost-govet/emptystrcmp"
+	"github.com/mattermost/mattermost-govet/emptyStrCmp"
 	"github.com/mattermost/mattermost-govet/equalLenAsserts"
 	"github.com/mattermost/mattermost-govet/immut"
 	"github.com/mattermost/mattermost-govet/inconsistentReceiverName"
@@ -27,6 +27,6 @@ func main() {
 		inconsistentReceiverName.Analyzer,
 		apiAuditLogs.Analyzer,
 		immut.Analyzer,
-		emptystrcmp.Analyzer,
+		emptyStrCmp.Analyzer,
 	)
 }

--- a/main.go
+++ b/main.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"github.com/mattermost/mattermost-govet/apiAuditLogs"
+	"github.com/mattermost/mattermost-govet/emptystrcmp"
 	"github.com/mattermost/mattermost-govet/equalLenAsserts"
 	"github.com/mattermost/mattermost-govet/immut"
 	"github.com/mattermost/mattermost-govet/inconsistentReceiverName"
@@ -26,5 +27,6 @@ func main() {
 		inconsistentReceiverName.Analyzer,
 		apiAuditLogs.Analyzer,
 		immut.Analyzer,
+		emptystrcmp.Analyzer,
 	)
 }

